hw/xbox/nv2a/nv2a.c

```diff
@@ -506,22 +511,11 @@ static void nv2a_realize(PCIDevice *dev, Error **errp)
                                     &d->block_mmio[i]);
     }
 
-    /* init fifo cache1 */
-    qemu_spin_init(&d->pfifo.cache1.alloc_lock);
-    qemu_mutex_init(&d->pfifo.cache1.cache_lock);
-    qemu_cond_init(&d->pfifo.cache1.cache_cond);
-    QSIMPLEQ_INIT(&d->pfifo.cache1.cache);
-    QSIMPLEQ_INIT(&d->pfifo.cache1.working_cache);
-    QSIMPLEQ_INIT(&d->pfifo.cache1.available_entries);
-    QSIMPLEQ_INIT(&d->pfifo.cache1.retired_entries);
-
```

Comment:
So you decided to do away with this optimization?

Reply:
Yes, because it was a hack. Hardware doesn't work this way. The way the hardware works is exposed via registers and is simpler.

It's also necessary to implement the limited cache size as D3D is able to patch the FIFO while its still being processed. If you process the entire FIFO in one step, then you might run into issues as the GPU is not aware of those patches.

Doing it properly will hopefully also make it possible to [use nv2a-trace](https://github.com/XboxDev/nv2a-trace) in the future.

Reply:
@PatrickvL Yeah, the `available_entries` and `retired_entries` queues is an optimization to the older code which was not HW accurate. That optimization did improve performance and is fine for now, but going forward we want to clean up the FIFO stuff altogether so it'll get thrown out with the rest of the cache entry stuff, and that's what this PR seeks to do.